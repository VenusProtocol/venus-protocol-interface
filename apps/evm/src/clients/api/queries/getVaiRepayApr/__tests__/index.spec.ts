import type { PublicClient } from 'viem';

import fakeAddress from '__mocks__/models/address';

import { getVaiRepayApr } from '..';

describe('getVaiRepayApr', () => {
  it('returns VAI repay APR in the right format on success', async () => {
    const fakePublicClient = {
      readContract: async () => 1000000000000000000n,
    } as unknown as PublicClient;

    const res = await getVaiRepayApr({
      publicClient: fakePublicClient,
      vaiControllerAddress: fakeAddress,
    });

    expect(res).toMatchSnapshot();
  });
});
