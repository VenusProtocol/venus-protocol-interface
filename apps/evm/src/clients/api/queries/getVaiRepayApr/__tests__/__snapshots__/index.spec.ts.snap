// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getVaiRepayApr > returns VAI repay APR in the right format on success 1`] = `
{
  "repayAprPercentage": "100",
}
`;
